MapboxCoreNavigation/EventsManager.swift

```diff
@@ -18,6 +18,8 @@ open class EventsManager: NSObject {
     
     weak var routeController: Router!
     
+    private var didSendCancelEvent = false
```

Comment:
I have concerns with moving this into the EventsManager

Reply:
I don't think we should make this component stateful in this way; it complicates potentially reusing it. Not sure if that's a use case but silently failing to send subsequent failures would not be good.

Reply:
The events manager is stateful, in particular, its `sessionState`. I fail to see how it was better when the RouteController handled the state of the events manager which definitely complicates potentially reusing it.

The only change here is that the EventsManager will handle `applicationWillTerminate(_:)` on its own instead of having the RouteController reporting to the EventsManager that `applicationWillTerminate(_:)`. It's self-contained and more reusable.

Reply:
rather than add a boolean to keep track of this state, we should key it off of the `SessionState`'s state transition.